app/javascript/lib/UseCases/Rondels/MoveToSlot.js

```diff
@@ -0,0 +1,62 @@
+import AvailableSlots from './SlotSelection/AvailableSlots';
+import SlotDistanceCosts from './SlotSelection/SlotDistanceCosts';
+
+const InvalidMoveError = class extends Error {
+  constructor(fromRondelSlot, toRondelSlot) {
+    super(`Attepting to make invalid move from ${fromRondelSlot?.id} to ${toRondelSlot.id}.`);
+
+    this.name = 'InvalidMoveError';
+  }
+};
+
+export default class MoveToSlot {
+  static get InvalidMoveError() {
+    return InvalidMoveError;
+  }
+
+  #availableSlots;
+  #costs;
+
+  constructor(game) {
+    this.#availableSlots = new AvailableSlots(game.rondel, game.availableFreeRondelSlotCount, game.availablePaidRondelSlotCount);
+    this.#costs = new SlotDistanceCosts(game);
+  }
+
+  get availableSlots() {
+    return this.#availableSlots;
+  }
+  get slotDistanceCosts() {
+    return this.#costs;
+  }
+
+  forceMoveNation(nation, toRondelSlot) {
+    if (nation.residingRondelSlot) {
```

Comment:
While we're on the subject of lint, Im not too keen on it forcing me to declare static methods either, but Im half willing to let it slide.

Usecases tend to be more volatile, and can easily be changed to need class variables on a dime (or vice-versa). Prematurely declaring it static causes more annoyances down the road should that happen imo, but it's not too bad to fix either since it's usually just search and replace

Reply:
[a07bfc6](https://github.com/Thrillberg/imperial/pull/731/commits/a07bfc69ce5c855a6936c4d64057169698395e72)